src/components/DiagramTab/DiagramTab.jsx

```diff
@@ -1,22 +1,56 @@
-//Ten komponent zawiera logikę subskrybowania na redux store, otrzymania wszystkich niezbędnych danych.
-//Implementacja metod filtrowania danych.
-//I te elementy stanu i metod przekazać przez props do podrzędnych elementów Table i Chart
-
-//Jest prawdopodobnie dzieckiem DashboardPage z którego pobiera dane i zapisuje je na redux w finance
-
+import { useSelector } from 'react-redux';
+import { selectTotalBalance, selectData } from '../../redux/finance/selectors';
+// import { setTotalBalance, setData, resetFinance } from '../../redux/finance/financeSlice';
+import { categories } from './diagramUtils';
 import Chart from '../Chart/Chart';
 import Table from '../Table/Table';
 import css from './DiagramTab.module.css';
 
 export const DiagramTab = () => {
+  
+  const totalBalance = useSelector(selectTotalBalance);
+  const data = useSelector(selectData);
+
+  console.log(data)
+
+  // reducing expenses values to single categories (totals by category)
+  const categorySum = data.reduce((acc, transaction) => {
+
+    const { category, amount } = transaction;
+    
+    if (!acc[category]) {
+      acc[category] = 0;
+    }
+    
+    acc[category] += amount;
+    
+    return acc;
+  }, {});
+
+  console.log(categorySum)
+
+  // populating transactions object based on array of categories in correct order
+  const transactions = categories.map(category => ({
+    category,
+    amount: categorySum[category] || 0,
+  }));
+
+
+  const totalIncome = data.map(e => e.isIncome ? e.amount : 0).reduce((acc, curr) => acc + curr, 0)
+  const totalExpenses = data.map(e => e.isIncome ? 0 : e.amount).reduce((acc, curr) => acc + curr, 0)
+  
+  console.log(transactions)
+
   return (
     <>
-      <div className={css.wrapper}>
-        <h3 className={css.diagramHeader}>Statistics</h3>
-        <div className={css.diagramContainer}>
-          <Chart />
-          <Table />
-        </div>
+    <div>
```

Comment:
ten div jest niezamknięty, nie wiem czy w ogóle jest potrzebny, tak samo jak czy potrzebny jest <> </>

Reply:
Popraw koniecznie bo wywala błąd krytyczny przy ładowaniu komponentu

Reply:
To nie jest mój kod. Konrad musiał to zmienić. Wywala błąd, bo komponenty nie mają propsów z danymi. Rozwiązywałem wczoraj tej konflikt. Nie wiem czemu zapisało nie moją wersję.

Reply:
tutaj nie chodzi o propsy tylko o to że jeden <div> jest niezamknięty

Reply:
Przede wszystkim chodzi o to, że to nie jest mój kod. Ja tam nawet nie mam takiego diva, w komponenty czekają na propsy. Musiał się źle konflikt rozwiązać. To są zmiany wprowadzone przez Konrada. Zaraz podeślę jak to u mnie wygląda i zobaczę jak to można podmienić.